### ISSUE TITLE
External storage - only showing first 15 files or folders

### ISSUE THREAD

**Issue by KaliszAd** [owncloud/core#19384]
I have this problem although using Owncloud 8.1.3 and MySQL 8.1.3 on a Debian 8 Jessie based server using external storage plugin to mount local folders (/mnt/raid). I noticed, only the first 15 files or folders (depenting on ascending or descending sorting) are shown. If the directory has less than 15 files/folders, all are shown. The page has length appropriate to the number of files/folders that _should_ be shown, it just isn't showing all files/folders.
There is no problem, when listing using Webdav.

## Further analysis:

The problem seems to be this (Firefox 40.0.3, Iceweasel 38)

    14:18:10.700 TypeError: OC.Share is undefined1 share.js:75:5

almost at the bottom of the page, where the directories should be, there is a small notice, there are 38 directories and 32 files waiting. It seems, 

Internet Explorer 11 displays the affected javascript lines:
```javascript
if (!OCA.Sharing.sharesLoaded){
                    OC.Share.loadIcons('file', fileList, function() {
                        // since we don't know which files are affected, just refresh them all
                        updateIcons();
                    });
```
The problem seems to be with the loadIcons() function. I mean, refreshing all files and directories is maybe not the best way from the efficiency point of view, besides not displaying the contents.

Setting up a diferent cron didn't help.

### For more context, please visit Issue #8710 or contact @PVince81

**Comment by PVince81** [comment]
`OC.Share is undefined` ???

Something must be wrong with your source code.
Line 75 of share.js in 8.1.3 is this: https://github.com/owncloud/core/blob/v8.1.3/core/js/share.js#L75

Maybe you've got a mix of old source code and new source code.

How did you upgrade ? (updater app, distro packages or manually)
You should delete the old source code first (and keep "data", "config") then extract the code from 8.1.3.

**Comment by KaliszAd** [comment]
I do have this version the whole time from the openSUSE binary repository. I reinstalled the package and the behavior changed slightly in the way, that I have 22 files/ folders visible now. I still get the problem in Iceweasel 38.2.1 (Firefox):

    Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery.min.js:1:0
    window.controllers is deprecated. Do not use it for UA detection. ace.js:1716:0
    Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. jquery.min.js:5:0
    TypeError: OC.Share is undefined share.js:75:5

**Comment by PVince81** [comment]
@KaliszAd did you disable sharing in the admin page ? Or did you disable the "Share files" app ?
You need to either enable both or disable both.

**Comment by PVince81** [comment]
This is the PR that fixes it for the upcoming 8.1.4: https://github.com/owncloud/core/pull/19439

If that doesn't solve this for you, feel free to reopen.

**Comment by KaliszAd** [comment]
Ok, this fixes it! Disabling the sharing app and sharing API in admin page is the solution, thank you!
